test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -9813,71 +9813,34 @@ def create_import_hook(self, name, attrs=None, body=None, overwrite=True,
-                                                % (hfile, m), interval=1)
-                                            self.log_match(
-                                                msg + " %s to %s"
-                                                % (pyfile, m), interval=1)
-                                            self.moms[m].log_match(
-                                                logmsg, starttime=t)
-                        else:
-                            return False
-            else:
-                if 'exec' in attrs['event']:
-                    hook_py = name + '.PY'
-                    hook_hk = name + '.HK'
-                    pyfile = os.path.join(self.pbs_conf['PBS_HOME'],
-                                          "server_priv", "hooks", hook_py)
-                    hfile = os.path.join(self.pbs_conf['PBS_HOME'],
-                                         "server_priv", "hooks", hook_hk)
-                    logmsg = hook_py
-                    logmsg += ";copy hook-related file request received"
-
-                    cmd = os.path.join(self.client_conf['PBS_EXEC'], 'bin',
-                                       'pbsnodes') + ' -a'
-                    cmd_out = self.du.run_cmd(self.hostname, cmd, sudo=True)
-                    if cmd_out['rc'] == 0:
-                        for i in cmd_out['out']:
-                            if re.match(r'\s+Mom = ', i):
-                                mom_names = i.split(' = ')[1].split(',')
-                                for m in mom_names:
-                                    if m in self.moms:
-                                        msg = "successfully sent hook file"
-                                        self.log_match(
-                                            msg + " %s to %s"
-                                            % (hfile, m), interval=1)
-                                        self.log_match(
-                                            msg + " %s to %s"
-                                            % (pyfile, m), interval=1)
-                                        self.moms[m].log_match(
-                                            logmsg, starttime=t)
-                    else:
-                        return False
-
+            for hook in events:
+                hook_py = name + '.PY'
+                hook_hk = name + '.HK'
+                pyfile = os.path.join(self.pbs_conf['PBS_HOME'],
+                                      "server_priv", "hooks", hook_py)
+                hfile = os.path.join(self.pbs_conf['PBS_HOME'],
+                                     "server_priv", "hooks", hook_hk)
+                logmsg = hook_py + ";copy hook-related file request received"
+                cmd = os.path.join(self.client_conf['PBS_EXEC'], 'bin',
+                                   'pbsnodes') + ' -a' + ' -Fjson'
+                cmd_out = self.du.run_cmd(self.hostname, cmd, sudo=True)
+                json_out = '\n'.join(cmd_out['out'])
```

Comment:
Put `'\n'.join(cmd_out['out'])` in `json.loads()` below, also add error check for run_cmd()